fem/fe/fe_nurbs.hpp

```diff
@@ -152,6 +153,326 @@ class NURBS3DFiniteElement : public NURBSFiniteElement
                              DenseMatrix &hessian) const;
 };
 
+
+/// An arbitrary order H(div)-conforming 2D NURBS element on a square
+class NURBS_HDiv2DFiniteElement : public VectorFiniteElement,
+   public NURBSFiniteElement
+{
+protected:
+   mutable Vector shape_x, shape_y, dshape_x, dshape_y, d2shape_x, d2shape_y;
+   mutable Vector shape1_x, shape1_y, dshape1_x, dshape1_y, d2shape1_x, d2shape1_y;
+   mutable Vector u;
+   mutable DenseMatrix du;
+   mutable Array <const KnotVector*> kv1;
+
+public:
+   /// Construct the NURBS_HDiv22DFiniteElement of order @a p
+   NURBS_HDiv2DFiniteElement(int p, int vdim)
+      : VectorFiniteElement(2, Geometry::SQUARE, 2*(p + 1)*(p + 2), p,
+                            H_DIV,FunctionSpace::Qk),
+        NURBSFiniteElement(2),
+        shape_x(p + 1), shape_y(p + 1), dshape_x(p + 1),
+        dshape_y(p + 1), d2shape_x(p + 1), d2shape_y(p + 1),
+        shape1_x(p + 2), shape1_y(p + 2), dshape1_x(p + 2),
+        dshape1_y(p + 2), d2shape1_x(p + 2), d2shape1_y(p + 2),
+        u(dof), du(dof,2)
+   {
+      orders[0] = orders[1] = p;
+      kv1.SetSize(dim);
+      kv1[0] = nullptr;
+      kv1[1] = nullptr;
+   }
+
+   /// Construct the NURBS_HDiv22DFiniteElement with x-order @a px and y-order @a py
```

Comment:
`22` -> `2` in many places in this file.

Reply:
Indeed! Somewhere even `3`.